### ISSUE TITLE
linter: `unicorn/no-array-sort` reports non-array `.sort()` calls

### ISSUE THREAD

**Issue by ardalanamini** [oxc-project/oxc#22487]
### What version of Oxlint are you using?

1.65.0

### What command did you run?

`oxlint -c ./oxlint.config.ts --type-aware .`

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "categories": {
    "correctness": "off"
  },
  "plugins": ["unicorn"],
  "rules": {
    "unicorn/no-array-sort": "error",
  }
}
```

## Description

`unicorn/no-array-sort` reports false positives for non-array `.sort()` methods, specifically Mongoose query sorting.

The rule is intended to discourage mutating `Array.prototype.sort()` and suggest `toSorted()` instead. However, Mongoose’s `.sort()` is not `Array#sort`; it is a query builder method used to specify database sort order.

## Reproduction

```ts
import mongoose from "mongoose";

const userSchema = new mongoose.Schema({
  name: String,
  createdAt: Date,
});

const User = mongoose.model("User", userSchema);

async function getUsers() {
  return User.find().sort({ createdAt: -1 });
}
```

## Actual behavior

Oxlint reports `unicorn/no-array-sort` on the Mongoose `.sort()` call:

```ts
User.find().sort({ createdAt: -1 });
            ^^^^
```

It suggests avoiding `Array#sort()` / using `toSorted()`.

## Expected behavior

The rule should not report this code, because `User.find().sort(...)` is not `Array.prototype.sort()`.

This `.sort()` call belongs to Mongoose’s query API and does not mutate a JavaScript array.

## Why this matters

In Node.js/TypeScript backend projects, `.sort()` is commonly used by database query builders and ORMs/ODMs, for example:

```ts
User.find().sort({ createdAt: -1 });
User.find().sort("-createdAt");
Post.find({ published: true }).sort({ updatedAt: "desc" });
```

Flagging these as `unicorn/no-array-sort` creates noisy false positives and makes the rule difficult to enable in projects using Mongoose.

## Expected scope of the rule

This should still be reported:

```ts
const items = [3, 1, 2];
items.sort();
```

But this should not be reported:

```ts
User.find().sort({ createdAt: -1 });
```

**Comment by ardalanamini** [comment]
Additional note: I’m seeing a similar false positive pattern with `unicorn/no-array-method-this-argument`.

For example, Mongoose query methods (e.g. `.find()`) can also be flagged even though the method being called is not an `Array.prototype` method and the receiver is not an array.

I’m not sure whether this should be tracked as a separate issue or whether both rules are affected by the same underlying assumption in the implementation: treating certain method names as array methods without proving that the receiver is actually an array.

If the fix for `unicorn/no-array-sort` is likely to be rule-specific, I can open a separate issue for `unicorn/no-array-method-this-argument` with a focused reproduction. But if this comes from shared rule logic or a broader lack of receiver/type validation in the Unicorn compatibility rules, it may make sense to handle them together.

**Comment by ardalanamini** [comment]
I noticed that the rule source already has a related `TODO` covering this class of false positives:

https://github.com/oxc-project/oxc/blob/625758ab25657c0fc13555c890aeca979be8ebdf/crates/oxc_linter/src/rules/unicorn/no_array_sort.rs#L156

The `TODO` examples seem to include non-array/query-builder-style `.sort(...)` calls such as `collection.sort({field: 1})`, `query.sort("field")`, and Mongo-style chained query sorting.

The Mongoose case appears to be the same underlying issue rather than a Mongoose-specific edge case:

```ts
User.find().sort({ createdAt: -1 });
```

From what I can tell, the rule currently treats every `.sort(...)` as `Array#sort()` based primarily on the static property name, without checking whether the receiver is actually an array or not.
